### DISCUSSION TITLE
Enquiry about differences between esmFold and AlphaFold2 in the predicted LDDT head

### DISCUSSION THREAD

**Discussion by hkotb** [facebookresearch/esm#381] [Q&A]
Hi, 
Thanks for sharing this wonderful work!

I was looking at the code, and I found that you are not using relu or any activation function between linear layers in some parts of the code. That doesn't match AlphaFold2 architecture in the absence of relu and the number of linear layers as well:

`
self.lddt_head = nn.Sequential(
     nn.LayerNorm(cfg.trunk.structure_module.c_s),
     nn.Linear(cfg.trunk.structure_module.c_s, cfg.lddt_head_hid_dim),
     nn.Linear(cfg.lddt_head_hid_dim, cfg.lddt_head_hid_dim),
     nn.Linear(cfg.lddt_head_hid_dim, 37 * self.lddt_bins),
)
`
As I know, chaining multiple linear layers without adding any non-linear components will make the chain worthless and can be replaced with a single linear layer.

Also, you compute the expected value of pLDDT based on all the atoms, while AlphaFold2 does the calculations based on alpha carbon atoms and masks the rest of the atoms, as I understood.

`        output["mean_plddt"] = (output["plddt"] * output["atom37_atom_exists"]).sum(
            dim=(1, 2)
        ) / output["atom37_atom_exists"].sum(dim=(1, 2))`

Did I understand what you're doing correctly? And if so, are you doing that on purpose? (omitting relu functions, changing number of layers, and calculating pLDDT based on all atoms)

Thanks!

**Comment by tomsercu** [comment]
Thanks for flagging these issues! We'll fix the lddt head in future versions of esmfold.

**Comment by DaRinker** [comment]
I was going to post a similar question regarding pLDDT differences with AF2. While I generally don't see a large range of pLDDT variation for ATOMS in any given reside, instances exist Here's a random example from an ESMfold prediction I just ran:
```
`ATOM     72  N   ILE A  10     -12.426  -0.957   9.516  1.00  0.91           N

ATOM     73  CA  ILE A  10     -11.259  -0.084   9.589  1.00  0.92           C
  
ATOM     74  C   ILE A  10     -10.045  -0.883  10.057  1.00  0.92           C
  
ATOM     75  CB  ILE A  10     -11.507   1.114  10.532  1.00  0.89           C 
 
ATOM     76  O   ILE A  10     -10.172  -1.791  10.882  1.00  0.91           O
  
ATOM     77  CG1 ILE A  10     -12.741   1.904  10.078  1.00  0.73           C 
 
ATOM     78  CG2 ILE A  10     -10.272   2.017  10.595  1.00  0.72           C  

ATOM     79  CD1 ILE A  10     -13.202   2.958  11.076  1.00  0.70          C   `
```

How should I best compare the ESMfold model to an AF model here?--average over all the atoms? Consider only the alpha carbon? Seems like a highly confident backbone prediction should be more meaningful that a much lower confident side chain prediction  (since that position will probalby change upon any post-modeling relaxation)
